### ISSUE TITLE
multiple {@link} jsDocs tag result in one big incorrect hyperlink

### ISSUE THREAD

**Issue by RaptorX** [thqby/vscode-autohotkey2-lsp#429]
Type: <b>Bug</b>

• Add more than one `{@link}` tag in a jsDoc comment

### Expected
I expect to see two different links

### Result
-- img --

### Notes
Seems like the regex for that is greedy and starts on one link and ends up with the bracket from the other link.

---

Extension version: 2.2.5
VS Code version: Code 1.84.0 (d037ac076cee195194f93ce6fe2bdfe2969cc82d, 2023-11-01T11:29:04.398Z)
OS version: Windows_NT x64 10.0.22621
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-4460  CPU @ 3.20GHz (4 x 3198)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.88GB (19.39GB free)|
|Process Argv|--crash-reporter-id 3e042e59-cb22-4371-968a-6f3e35abe666|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscorecescf:30445987
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
pythonvs932:30410667
py29gd2263:30880072
vsclangdc:30486549
c4g48928:30535728
dsvsc012:30540252
pynewext54:30695312
azure-dev_surveyone:30548225
3biah626:30602489
f6dab269:30613381
showlangstatbar:30737416
pythonfmttext:30731395
fixshowwlkth:30771522
showindicator:30805244
pythongtdpath:30769146
i26e3531:30792625
pythonnosmt12:30797651
pythonidxpt:30866567
pythonnoceb:30805159
dsvsc013:30795093
dsvsc014:30804076
dsvsc015:30845448
pythontestfixtcf:30871695
pythonregdiag2:30871582
pythonmypyd1:30879173
pythoncet0:30874138
pythontbext0:30879054
dsvsc016cf:30880770

```

</details>

**Comment by SaifAqqad** [comment]
@RaptorX does it actually link correctly to the method/property specified?

In my testing, it kept linking it as a normal URI so vscode tries to open it as a file
jsdoc:
 ![image](https://github.com/thqby/vscode-autohotkey2-lsp/assets/47293197/9475621b-8ca2-4f01-a9ff-d23cddd2c38a)
doc overlay: 
![image](https://github.com/thqby/vscode-autohotkey2-lsp/assets/47293197/3af27a9b-c602-432c-ae06-dac311db6c4e)
when clicking the link: 
![image](https://github.com/thqby/vscode-autohotkey2-lsp/assets/47293197/eb198627-955f-4e7b-a25a-7a62dc701686)

**Comment by RaptorX** [comment]
It behaves as follows:

• individual links on their own line try to trigger the correct link
• multiple links on the same line highlight the incorrect text and try to open the **first** link only
• the second link on multi link lines is never accessible

here is some test code:

```
/**
	 * {@link SQLite3.open_v2 test1} this is a test
	 * 
	 * {@link SQLite3.close_v2 test2} this is a test
	 * 
	 * {@link SQLite3.open_v2 test1} {@link SQLite3.close_v2 test2}
	 * 
	 * {@link SQLite3.open_v2 test1} this is a test {@link SQLite3.close_v2 test2}
	 */
	Test()
	{
		return SQLite3.open_v2()
	}
```

**Comment by SaifAqqad** [comment]
but does the rendered link in the first case (only a single link) even work, does it navigate you to `SQLite3.open_v2`?

**Comment by RaptorX** [comment]
> It behaves as follows:
>  • multiple links on the same line highlight the incorrect text and try to open the **first** link only
> • the second link on multi link lines is never accessible


Yes. As I mentioned in my previous comment, the link tries to open the first one always, in this example `SQLite.open_v2`

**Comment by RaptorX** [comment]
Yes, as I commented above if the links are by themselves on their lines, they behave as expected: they try to open their correct linked value showing the error you sent above.

**Comment by RaptorX** [comment]
> but does the rendered link in the first case (only a single link) even work, does it navigate you to `SQLite3.open_v2`?

They shouldn't navigate to a file unless you use JSDOC correctly and create you documentation files.

Those links are not meant to open in the current file if that's what you are expecting.

**Comment by thqby** [comment]
```
/**
 * {@link @cls.method} is same as {@link cls.prototype.method}
 */
fn() {
	cls().method()
}

class cls {
	method() {
		
	}
}
```
Added a link to go to the specified symbol.
